watertap/property_models/multicomp_aq_sol_prop_pack.py

```diff
@@ -370,6 +402,14 @@ def build(self):
             units=pyunits.dimensionless,
             doc="Ion charge",
         )
+        self.diffus_phase_comp_param = Param(
```

Comment:
Do we actually need this param? It looks like it isn't used anywhere, except when referencing diffusivity in unit models, which I think doesn't have to be the case. The reason is that you create the `diffus_phase_comp` variable and fix it when `DiffusivityCalculation` is `None`.

I would suggest removing this Param, unless i am missing a good reason for its inclusion. Originally I said that we could later on update `diffus_phase_comp` to be a variable that gets fixed instead of a parameter, but line 1329 already does the job. So, I would suggest removing this Param, unless i am missing a good reason for its inclusion.

Reply:
Initially I removed-  The absence of it created some errors where a unit call this param by `properties.params.diffus_phase_comp` directly (as against calling it by some state_block).  One example is in the nanofiltration file (where this param is retained, while I modified those in the ED files).   So I added this param back and thought it might not be a bad idea to have it staying out of state blocks for easy calling when needed.

Reply:
Hmmm... If that is the case, I would prefer just declaring `diffus_phase_comp` as a variable here and fixing it, when DiffusivityCalculation is None. Then in the method, you can add object reference and/or do Constraint.Skip when calculation type if none (or something like that--didn't think it through).

Reply:
That is already implemented.  That's why the param now has a different name.  If you wanna remove the param, could you go ahead to modify the nanofiltration file to let the diffusivity called by the accurate `state_block` where applicable?   You can also simply delete `diffus_phase_comp_param` along as way.

Reply:
Honestly I like the way it is so sometimes a unit can call a constant diffusivity without bothering to decide which state_block to put it on.  Such a situation is often encountered.   If we remove this param completely, I don't see why we keep all the other parameters especially if "consistency" is so liked.

Reply:
> That is already implemented. That's why the param now has a different name. If you wanna remove the param, could you go ahead to modify the nanofiltration file to let the diffusivity called by the accurate `state_block` where applicable? You can also simply delete `diffus_phase_comp_param` along as way.

I meant for it to be implemented here on the parameter block, not where it is now.

Reply:
The way I am suggesting would allow the user to just call the `params.diffus_phase_comp` instead of pointing to a state block _or_ pointing to `params.diffus_phase_comp_param`. In a subsequent PR, we can convert other parameters to variables that get fixed on the parameter block, but I didn't want to hold up the PR much longer.